src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/ApplicationRepository.cs

```diff
@@ -83,22 +80,31 @@ public async Task<List<Application>> FindByOrg(string org)
         public async Task<Application> FindOne(string appId, string org)
         {
             string cosmosAppId = AppIdToCosmosId(appId);
-            Uri uri = UriFactory.CreateDocumentUri(DatabaseId, CollectionId, cosmosAppId);
 
             if (!_memoryCache.TryGetValue(appId, out Application application))
             {
-                application = await Client
-                .ReadDocumentAsync<Application>(
-                    uri,
-                    new RequestOptions { PartitionKey = new PartitionKey(org) });
+                try
+                {
+                    application = await Container.ReadItemAsync<Application>(cosmosAppId, new PartitionKey(org));
+                }
+                catch (CosmosException e)
```

Comment:
With changes like this we can remove most if not all exception handling in the controllers.

Reply:
Removed all exception handling from controllers